src/graphql/root/subscription/realtime-price.ts

```diff
@@ -0,0 +1,165 @@
+
+const RealtimePriceInput = GT.Input({
+  name: "RealtimePriceInput",
+  fields: () => ({
+    currency: {
+      type: GT.NonNull(DisplayCurrencyGT),
+      defaultValue: DisplayCurrency.Usd,
+    },
+  }),
+})
+
+type RealtimePriceSubscribeArgs = {
+  input: {
+    currency: DisplayCurrency | Error
+  }
+}
+
+const RealtimePriceSubscription = {
+  type: GT.NonNull(RealtimePricePayload),
+  description: "Returns the price of 1 satoshi",
+  args: {
+    input: { type: GT.NonNull(RealtimePriceInput) },
+  },
+  resolve: async (
+    source:
+      | {
+          errors: IError[]
+          timestamp?: Date
+          centsPerSat?: number
+          centsPerUsdCent?: number
+          displayCurrency?: DisplayCurrency
+        }
+      | undefined,
+    args: RealtimePriceSubscribeArgs,
+  ) => {
+    if (source === undefined) {
+      throw new UnknownClientError({
+        message:
+          "Got 'undefined' payload. Check url used to ensure right websocket endpoint was used for subscription.",
+        level: "fatal",
+        logger: baseLogger,
+      })
+    }
+
+    const { errors, timestamp, centsPerSat, centsPerUsdCent, displayCurrency } = source
+    if (errors) return { errors: errors }
+    if (!timestamp || !centsPerSat || !centsPerUsdCent || !displayCurrency) {
+      return { errors: [{ message: "No price info" }] }
+    }
+
+    const { currency } = args.input
+    if (currency instanceof Error) throw currency
+
+    if (displayCurrency !== currency) {
+      throw new UnknownClientError({
+        message: "Got 'invalid' payload.",
+        level: "fatal",
+        logger: baseLogger,
+      })
+    }
+
+    return {
+      errors: [],
+      realtimePrice: {
+        timestamp: new Date(timestamp),
+        denominatorCurrency: currency,
+        btcSatPrice: {
+          base: Math.round(centsPerSat * 10 ** SAT_PRICE_PRECISION_OFFSET),
+          offset: SAT_PRICE_PRECISION_OFFSET,
+          currencyUnit: `${currency}CENT`,
+        },
+        usdCentPrice: {
+          base: Math.round(centsPerUsdCent * 10 ** USD_PRICE_PRECISION_OFFSET),
+          offset: USD_PRICE_PRECISION_OFFSET,
+          currencyUnit: `${currency}CENT`,
+        },
+      },
+    }
+  },
+  subscribe: async (_: unknown, args: RealtimePriceSubscribeArgs) => {
+    const { currency: displayCurrency } = args.input
+
+    const immediateTrigger = customPubSubTrigger({
+      event: PubSubDefaultTriggers.PriceUpdate,
+      suffix: crypto.randomUUID(),
+    })
+
+    if (displayCurrency instanceof Error) {
+      pubsub.publishImmediate({
+        trigger: immediateTrigger,
+        payload: { errors: [{ message: displayCurrency.message }] },
+      })
+      return pubsub.createAsyncIterator({ trigger: immediateTrigger })
+    }
+
+    const currencies = await Prices.listCurrencies()
+    if (currencies instanceof Error) {
+      pubsub.publishImmediate({
+        trigger: immediateTrigger,
+        payload: { errors: [mapError(currencies)] },
```

Comment:
Can use this alternative method to also get the `code` property

```suggestion
        payload: { errors: [mapAndParseErrorForGqlResponse(currencies)] },
```

Reply:
Updated https://github.com/GaloyMoney/galoy/pull/2298/commits/8b2fe7e6169dcf8f3eecfc25ad65f54d1e53a66c